crypto/bio/bss_mem.c

```diff
@@ -241,29 +250,36 @@ static long mem_ctrl(BIO *b, int cmd, long num, void *ptr)
     BIO_BUF_MEM *bbm = (BIO_BUF_MEM *)b->ptr;
     BUF_MEM *bm;
 
+    if (b->flags & BIO_FLAGS_MEM_RDONLY)
+        bm = bbm->buf;
+    else
+        bm = bbm->readp;
+
     switch (cmd) {
     case BIO_CTRL_RESET:
         bm = bbm->buf;
         if (bm->data != NULL) {
-            /* For read only case reset to the start again */
-            if ((b->flags & BIO_FLAGS_MEM_RDONLY) || (b->flags & BIO_FLAGS_NONCLEAR_RST)) {
-                bm->length = bm->max;
+            /* For read only case just reset to the start again */
+            if (!(b->flags & BIO_FLAGS_MEM_RDONLY)) {
```

Comment:
if I see that right, in this case bm == bbm->readp
and therefore all changes to bm are overwritten below except the memset possibly?

Reply:
No, it was corrected in the later fixup commit. The bm==bbm->buf always in the CTRL_RESET.